### PR TITLE
[TDX verification] [rough PR] Overview of changes required to make TDX verification work in OE

### PR DESCRIPTION
**Pull request by RRathna**
Signed-off-by: Rathna Ramesh <rathna1993@gmail.com>

### CODE REVIEW

File: host/sgx/sgxquote.c [hunk 1]
```diff
@@ -1068,6 +1088,84 @@ static oe_result_t sgx_qvl_error_to_oe(quote3_error_t error)
...
+    else
+    {
+        OE_LOG_ERROR(
+            "tdx_qv_get_quote_supplemental_data_size failed: 0x%04x", dcap_ret);
+        supplemental_data_size = 0;
+        OE_RAISE(
+            OE_SGX_QUOTE_LIBRARY_ERROR); // Should define a better error code
+    }
+
+    validation_time = convert_oe_datetime_to_time_t(input_validation_time);
```
Comment (RRathna) [inline host/sgx/sgxquote.c:1136]: To-Do: implement this

File: common/sgx/quote.c [hunk 2]
```diff
@@ -454,6 +459,24 @@ oe_result_t oe_verify_sgx_quote(
     if (quote == NULL)
         OE_RAISE(OE_INVALID_PARAMETER);
 
+    // Check if TDX quote
+    uint32_t quote_type =
```
Comment (RRathna) [inline common/sgx/quote.c:463]: [SGXDataCenterAttestationPrimitives/sgx_dcap_quoteverify.cpp at master · intel/SGXDataCenterAttestationPrimitives (github.com)](https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteVerification/dcap_quoteverify/sgx_dcap_quoteverify.cpp#L838)

File: host/sgx/sgxquote.c [hunk 3]
```diff
@@ -1068,6 +1088,84 @@ static oe_result_t sgx_qvl_error_to_oe(quote3_error_t error)
...
+        OE_LOG_ERROR(
+            "tdx_qv_verify_quote failed: 0x%04x, %s",
+            dcap_ret,
+            get_quote3_error_t_string(dcap_ret));
+        OE_RAISE(OE_SGX_QUOTE_LIBRARY_ERROR); // define better error code
+    }
+
+    // We might need to check `quote_verification_result` in detail for more
+    // info and bubbleup relevant return code
+    result = quote3_to_oe_error_codes_mapping(
```
Comment (RRathna) [inline host/sgx/sgxquote.c:1163]: Input to this function should have &collateral_expiration_status, &quote_verification_result, dcap_ret

**Comment by mingweishih** [comment]
Closing this for now.

**Pull request closed without merging**